src/main/java/io/debezium/connector/spanner/config/validation/ConnectionValidator.java

```diff
@@ -56,10 +56,7 @@ public ConnectionValidator validate() {
         if (!FieldValidator.isSpecified(googleCredentials) && !FieldValidator.isSpecified(credentialPath) && !FieldValidator.isSpecified(credentialJson)) {
             String message = String.format(PLEASE_SPECIFY_CONFIGURATION_PROPERTY_MSG, SPANNER_CREDENTIALS_PATH.name(),
                     SPANNER_CREDENTIALS_JSON.name(), GOOGLE_APPLICATION_CREDENTIALS_ENV_VAR);
-            LOGGER.error(message);
-            context.error(message, SPANNER_CREDENTIALS_PATH, SPANNER_CREDENTIALS_JSON);
-            result = false;
```

Comment:
Why did you remove false here?

Reply:
When using workload identity authentication, it is not necessary to set an authentication key. Such authentication methods must be supported.
https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to

Reply:
https://cloud.google.com/docs/authentication/provide-credentials-adc
> Note: Service account keys are a security risk if not managed correctly. You should[ choose a more secure alternative to service account keys](https://cloud.google.com/docs/authentication#auth-decision-tree) whenever possible. If you must authenticate with a service account key, you are responsible for the security of the private key and for other management operations such as [key rotation](https://cloud.google.com/iam/docs/key-rotation). For more information, see [best practices for managing service account keys](https://cloud.google.com/iam/docs/best-practices-for-managing-service-account-keys).

Reply:
Sorry, I think it's best here then to move the validation to check for the application default credentials here if the JSON key doesn't exist. Then, it's best to validate the new logic with a new unittest. Thanks.

Reply:
OK, I understand what you want to do. If the key is not set in the validation, how about calling `ServiceAccountCredentials.getApplicationDefault()` and checking the application default credentials setting?
If `IOException` is raised, I think we can make it a validation error.

Reply:
I think one of my proposals was lost above. How about chaning the validation logic this way

- if none of Google credentials, credential path and credential JSON is set the log INFO message tha application default credentials will be used
- if at least one but not all are set then emit the current validation error

@nancyxu123 WDYT?

Reply:
Hi Jiri! I'm a bit confused here, could I ask for a quick clarification?

My understanding was that at least one of Google credentials, credential path, credentials JSON, or application default credentials should be set. 

If none of them are set, we should emit an error. Otherwise, we should use one of them. Is that correct? @laughingman7743

Reply:
Yes, if the JSON key is not set, check the ApplicationDefaultCredentials and make an error. Is the following implementation acceptable?
https://github.com/debezium/debezium-connector-spanner/pull/65/commits/99e9087a34455baee7666fb1ae92a24118ab6294

Reply:
@nancyxu123 I am sorry you are right, I just misread the condition :-(.

Reply:
This looks great, thank you so much! @laughingman7743 @jpechane

Reply:
https://github.com/debezium/debezium-connector-spanner/pull/65/commits/5773cd89ab6788eab6c1be150d8cd79de2d3e098
I have added some test cases. Please check them.

For some reason the old versions of mockito-core and mockito-inline were used, so I explicitly added them to the dependencies. (Older versions cannot use mockStatic.)
https://github.com/debezium/debezium-connector-spanner/pull/65/commits/5773cd89ab6788eab6c1be150d8cd79de2d3e098#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R191-R200

Reply:
I think we may as well add a ConnectionValidator test for when the environment variable `GOOGLE_APPLICATION_CREDENTIALS` is present.
However, if we want to easily change the environment variables with tests, it should be better to add libraries such as SystemRule, SystemLambda, SystemStubs or JUnitPioneer.
https://stackoverflow.com/questions/8168884/how-to-test-code-dependent-on-environment-variables-using-junit
https://github.com/stefanbirkner/system-rules
https://github.com/stefanbirkner/system-lambda/
https://github.com/webcompere/system-stubs
https://github.com/junit-pioneer/junit-pioneer
If you are willing to add the libraries I would also add the environment variable tests.

Reply:
@laughingman7743 I don't think this is necessary to test it that much.